File: hashnode.py
import requests

graphql_endpoint = "https://gql.hashnode.com/"

headers = {
    'Content-Type': 'application/json',
    'Authorization': '325768d0-75b4-45e1-be2c-c60043311de6'
}

mutation_query = '''
mutation PublishPost($input: PublishPostInput!) {
  publishPost(input: $input) {
    post {
      id
      slug
      title
      subtitle
    }
  }
}
'''

mutation_variables = {
    "input": {
        "title": "abcdefcgh",
        "subtitle": "abcdefgh",
        "contentMarkdown": "abcdefgh",
        "coverImageOptions": {
            "coverImageURL": "https://2.img-dpreview.com/files/p/E~C33x0S928x522T1200x675~articles/7395606096/Google-Photos_1.jpeg",
            "isCoverAttributionHidden": False,
            "coverImageAttribution": "abc123",
            "coverImagePhotographer": "xyz789"
        },
        "slug": "xyz789",
        "tags": [{
            "name": "JavaScript",
            "slug": "javascript",
            "id": "56744721958ef13879b94cad"
        }],
        "disableComments": False,
        "publicationId": "65102296cee41f4569292580"
    }
}

# Make the GraphQL request
response = requests.post(graphql_endpoint, headers=headers, json={
                         'query': mutation_query, 'variables': mutation_variables})

# Check if request was successful and print response
if response.status_code == 200:
    print(response.json())
else:
    print("GraphQL request failed with status code:", response)

File: Hashnode.js
import fetch from 'node-fetch';
import dotenv from 'dotenv'

dotenv.config()

export class HashNode{
    constructor(Authorization, publicationId) {
        this.publicationId = publicationId
        this.graphqlEndpoint = "https://gql.hashnode.com/"
        this.headers={
            'Content-Type': 'application/json',
            'Authorization': Authorization
        };
        this.mutationQuery = `
            mutation PublishPost($input: PublishPostInput!) {
                publishPost(input: $input) {
                    post {
                        id
                        slug
                        title
                        subtitle
                    }
                }
            }`;
    }

    async createPost(title, subtitle, contentMarkdown, tags, coverImageURL) {
        const mutationVariables = {
            "input": {
                "title": title,
                "subtitle": subtitle,
                "contentMarkdown": contentMarkdown,
                "coverImageOptions": {
                    "coverImageURL": coverImageURL
                },
                "tags": tags,
                "disableComments": false,
                "publicationId": this.publicationId
            }
        
        };
        fetch(this.graphqlEndpoint, {
            method: 'POST',
            headers: this.headers,
            body: JSON.stringify({ query: this.mutationQuery, variables: mutationVariables })
        })
        .then(response => {
            if (response.ok) {
                return response.json();
            } else {
                throw new Error(`GraphQL request failed with status code: ${response.status}`);
            }
        })
        .then(data => {
            console.log(data);
        })
        .catch(error => {
            console.error(error);
        });
        
    }
}

// const hashnode = new HashNode(process.env.HASHNODE_KEY, process.env.PUBLICATION_ID);
// let test= await hashnode.createPost(
//     "testHash",
//     "subtitle of testhash",
//     "lets start the journey",
//     "https://images.unsplash.com/photo-1522202176988-66273c2fd55f?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=M3w1ODMzMjl8MHwxfHNlYXJjaHw0fHxjb2RpbmclMjBvbiUyMGxhcHRvcHxlbnwwfDB8fHwxNzExNDY5NjcxfDA&ixlib=rb-4.0.3&q=80&w=1080"
// )
